### ISSUE TITLE
Asynchronous requests

### ISSUE THREAD

**Issue by pencil** [remiprev/her#143]
We have some use cases where we fetch multiple resources for a single page request. To fetch this data in parallel we used some "fork-join black magic".

It would be nice to have the possibility to perform requests asynchronously. Examples:

``` ruby
User.all do |users|
  puts "#{users.length} users"
end

User.find(1) do |user|
  puts "Name: #{user.fullname}"
end

User.create(fullname: "Tobias Fünke") do |user|
  puts "User #{user.fullname} created"
end

@user = User.new(fullname: "Tobias Fünke")
@user.occupation = "actor"
@user.save do |user|
  puts "User #{user.fullname} updated"
end
```

When a `block` is present, requests are automatically performed asynchronously.
